src/udfs/depth_estimator.py

```diff
@@ -0,0 +1,135 @@
+from typing import List, Tuple
+
+import numpy as np
+import torchvision
+import sys
+import os
+import torch
+import cv2
+
+from src.model import net
+from src.udfs.abstract_udfs import AbstractClassifierUDF
+from src.models import FrameBatch, Prediction, FrameInfo, Point, BoundingBox, ColorSpace
+from torch.autograd import Variable
+from torchvision import transforms
+from src.depth_estimation_result import DepthEstimationResult
+from src.utils.frame_filter_util import FrameFilter
+
+sys.path.append('../')
+sys.path.append('.')
+folder = os.path.dirname(os.path.abspath(__file__))
+
+CMAP = np.load(os.path.join(folder, 'cmap_kitti.npy'))
+IMG_SCALE = 1. / 255
+IMG_MEAN = np.array([0.485, 0.456, 0.406]).reshape((1, 1, 3))
+IMG_STD = np.array([0.229, 0.224, 0.225]).reshape((1, 1, 3))
+HAS_CUDA = torch.cuda.is_available()
+NUM_CLASSES = 6
+NUM_TASKS = 2  # segm + depth
+
+class DepthEstimator(AbstractClassifierUDF):
+    """
+
+    UDF class for depth estimation feature. responsible for taking batch of frames.
+    It processes the frames, sends them to deep learning model for predictions.
+    Consolidates the model output and store them in DepthEstimationResult object.
+    Arguments:
+        Takes AbstractClassifierUDF as input since it implements the abstract UDF class.
+
+    """
+
+    @property
+    def name(self) -> str:
+        return "Net"
+
+    def __init__(self):
+        super().__init__()
+        self.model = net(num_classes=NUM_CLASSES, num_tasks=NUM_TASKS)
+        if HAS_CUDA:
+            _ = self.model.cuda()
+            _ = self.model.eval()
+        ckpt = torch.load(os.path.join(folder, 'ExpKITTI_joint.ckpt'))
+        self.model.load_state_dict(ckpt['state_dict'])
+

```

Comment:
remove the hard-coded folder and required data paths. Assume these will be passed as parameters